src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -376,26 +374,18 @@ public void moveToDefaultDirectoryAndRename() {
     }
 
     /**
-     * Asks the user for confirmation that he really wants to the delete the file from disk (or just remove the link).
      *
      * @return true if the linked file should be removed afterwards from the entry (i.e because it was deleted
      * successfully, does not exist in the first place or the user choose to remove it)
      */
-    public boolean delete() {
-        Optional<Path> file = linkedFile.findIn(databaseContext, preferences.getFilePreferences());
+    public boolean delete(Optional<ButtonType> buttonType, ButtonType removeFromEntry, ButtonType deleteFromEntry) {
+        Optional<Path> file = getDir();
```

Comment:
Please avoid passing Optionals as arguments.
Why at all is a ButtonType object being used in the ViewModel?
This does not seem right at all. The ViewModel should not have to deal with the ui elements, this breaks the mvvm pattern. Please use primitives here.